lib/utils.js

```diff
@@ -54,34 +56,39 @@ function checkAvailable(host, port, callback) {
  * @param {number} port the port to check for availability
  * @param {object} [options] optional settings
  * @param {number} [options.retryMS] the amount of time to wait between retry attempts in ms
+ * @param {number} [options.initialMS] the amount of time to wait before first attempting in ms
  * @param {number} [options.retryCount] the number of times to attempt retry
- * @param {function} callback
  */
-function waitForAvailable(host, port, options, callback) {
-  if (typeof options === 'function') (callback = options), (options = {});
-  options = Object.assign(
-    {},
-    {
-      retryMS: 100,
-      retryCount: 10
-    },
-    options
-  );
-
-  let count = options.retryCount;
-
-  function run() {
-    checkAvailable(host, port, (err, available) => {
-      if (err) return callback(err);
-
-      count--;
-      if (available) return callback();
-      if (count === 0) return callback(new Error('Server is unavailable'));
-      setTimeout(() => run(), options.retryMS);
-    });
-  }
+function waitForAvailable(host, port, options) {
+  return co(function*() {
+    options = Object.assign(
+      {},
+      {
+        initialMS: 300,
+        retryMS: 100,
+        retryCount: 25
```

Comment:
For the `retryCount`, what do you think is a good number for this?

The Linux errors I was receiving could have been unrelated which is why I originally had it at 100. With a `retryCount` of 100, that is a maximum wait time of 10.3s. At 25, the wait time is a maximum of 2.8s.

The bottleneck here is we want enough time for the server to get setup and throw errors. I don't have a lot of experience with MongoDB, but 2.8s seems like enough time as an average?

Reply:
I think 25 is fine.